From 4bd8cad6c61204fef5f8d1a42e2ec71d5ec0c46b Mon Sep 17 00:00:00 2001
From: Aziz Sellami <aziz.sellami@nxp.com>
Date: Tue, 7 Jan 2025 17:44:05 +0100
Subject: [PATCH 29/30] dts: imx95-15x15: add Harpoon industrial device tree

Add Harpoon industrial device tree for i.MX 95 15x15 EVK.
Reuse harpoon device tree:
 - Disable lpuart3 and mu3 on Linux for usage in inmate cell.
 - Use imx95 rpmsg dtsi for RPMsg usage to communicate between Root and inmate cell.
Disable reg_can2_stby, to be able to set it at runtime through scripts.

Signed-off-by: Aziz Sellami <aziz.sellami@nxp.com>
---
 arch/arm64/boot/dts/freescale/Makefile        |  2 +-
 .../imx95-15x15-evk-harpoon-industrial.dts    | 25 +++++++++++++++++++
 2 files changed, 26 insertions(+), 1 deletion(-)
 create mode 100644 arch/arm64/boot/dts/freescale/imx95-15x15-evk-harpoon-industrial.dts

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index 61b75fe0ad16..cff479932cd5 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -459,7 +459,7 @@ dtb-$(CONFIG_ARCH_MXC) += imx91-9x9-qsb-i3c.dtb
 dtb-$(CONFIG_ARCH_MXC) += imx95-15x15-evk.dtb
 dtb-$(CONFIG_ARCH_MXC) += imx95-15x15-evk-mqs.dtb
 dtb-$(CONFIG_ARCH_MXC) += imx95-15x15-evk-aud-hat.dtb
-dtb-$(CONFIG_ARCH_MXC) += imx95-15x15-evk-root.dtb imx95-15x15-evk-inmate.dtb imx95-15x15-evk-harpoon.dtb
+dtb-$(CONFIG_ARCH_MXC) += imx95-15x15-evk-root.dtb imx95-15x15-evk-inmate.dtb imx95-15x15-evk-harpoon.dtb imx95-15x15-evk-harpoon-industrial.dtb
 dtb-$(CONFIG_ARCH_MXC) += imx95-15x15-ab2.dtb
 dtb-$(CONFIG_ARCH_MXC) += imx95-15x15-evk-i2c-spi-slave.dtb
 DTC_FLAGS_imx95-15x15-evk-boe-wxga-lvds0-panel := -@
diff --git a/arch/arm64/boot/dts/freescale/imx95-15x15-evk-harpoon-industrial.dts b/arch/arm64/boot/dts/freescale/imx95-15x15-evk-harpoon-industrial.dts
new file mode 100644
index 000000000000..fc2caf693bfe
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/imx95-15x15-evk-harpoon-industrial.dts
@@ -0,0 +1,25 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright 2024 NXP
+ */
+
+#include "imx95-15x15-evk-root.dts"
+#include "imx95-rpmsg-ca55.dtsi"
+
+&lpuart3 {
+	/* To use this, needs dedicated SM to allow RTOS access lpuart3 and lpuart3 pinmux */
+	status = "disabled";
+};
+
+/* This Message Unit is used for communication between the RTOS cell and the SM */
+&mu3 {
+	status = "disabled";
+};
+
+&flexcan2 {
+	status = "disabled";
+};
+
+&reg_can2_stby {
+	status = "disabled";
+};
-- 
2.34.1

